### DISCUSSION TITLE
Bug in constructor of Image.php

### DISCUSSION THREAD

**Discussion by TutanRamon** [spatie/image#225] [General]
On line 31 of Image.php, there is a reference to ImagickDriver. However, if I choose to use a different imageDriver, this line is still loaded. I don't have ImagickDriver installed, so  I keep getting a class not found error. I don't have enough knowledge to fix this problem (I changed it to new GdDriver() for now, but I should be dynamic.

**Comment by lukasleitsch** [comment]
You can change the image drive before loading the image:

```php
\Spatie\Image\Image::useImageDriver(ImageDriver::Gd)
    ->loadFile('example.jpg')
 ```

**Reply by TutanRamon** [reply]
I know, but since the constructor of Image.php has a hard reference to imagick, the use of useImageDriver doesn't have any effect. So it seems.

**Reply by lukasleitsch** [reply]
The static method `useImageDriver` creates a new instance of the Image class. In the next step, the default image driver will be overwritten.

https://github.com/spatie/image/blob/main/src/Image.php#L67

The `useImageDriver` method is the official way to change the image driver: https://github.com/spatie/image/blob/main/docs/usage/basic-usage.md#selecting-a-driver

**Reply by TutanRamon** [reply]
I know that, but if you don't have the ImagickDriver installed on your system, you will get an exception when the model is created. Probably most developers have both drivers installed so you won't notice the exception.

**Reply by pauloacosta** [reply]
use Spatie\Image\Image;

Image::useImageDriver(ImageDriver::Gd)->loadFile($fullFilename)
                ->manualCrop($this->width, $this->height, $this->x, $this->y)
                ->width(100)
                ->height(100)
                ->save();

I am receiving: 

Class "App\Livewire\User\ImageDriver" not found

version:

 "spatie/image": "^3.8"
 
 Solution, import using:

 use Spatie\Image\Image; 
 use Spatie\Image\Enums\ImageDriver as ImageDriver;

**Comment by websitevirtuoso** [comment]
@lukasleitsch 

I am getting the same error
public static function load(string $pathToImage): static
 ```
    {
        if (! file_exists($pathToImage)) {
            throw CouldNotLoadImage::fileDoesNotExist($pathToImage);
        }

        return new static($pathToImage);
    }
    
 ```
    
    I have this code
    ```
    Image::useImageDriver(ImageDriver::Gd)
            ->load($tmpFilePath)->format(ImageExt::WEBP)
            ->quality(100)->optimize()->save($convertedImage);
    ```
According to 1 snipet. it doesn't use class properly imageDriver

According this documentation it should be
https://spatie.be/docs/image/v3/upgrading#content-working-with-the-image-object

My offre is fix in class, or create config for lib. To be able to change Driver via config.

**Comment by websitevirtuoso** [comment]
I could make it work but instead of load I had to use loadFile. 
![image](https://github.com/spatie/image/assets/5112918/24bdef7f-7c72-4056-bddf-18de89dce969)

Maybe error in documentation

**Reply by Fluotonic** [reply]
There is indeed an error in the documentation! I just replaced "loadFile" with "load" and VOILÀ !!!
Thanks @websitevirtuoso :-)

**Reply by Fluotonic** [reply]
OK actually it turns out I was mistaken... this worked well locally but it's because Imagick is available. It's still broken online...

**Comment by blushift80** [comment]
Hello,
I fixed the issue by replacing line 31 in the constructor:

$this->imageDriver = new ImagickDriver;

with

$this->imageDriver = new GdDriver;

This is because the GD library is an extension that is always present in a PHP installation, whereas Imagick is not. As a result, even if the driver is changed using the static method useImageDriver, the constructor still throws an exception in PHP installations that do not have Imagick.

Can this be fixed in the next release?

Thanks.

**Reply by freekmurze** [reply]
Feel free to submit a PR for this 👍

**Comment by andreasnij** [comment]
This should now be fixed in version 3.8.7 (https://github.com/spatie/image/releases/tag/3.8.7).
